compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -242,8 +242,9 @@ object Types {
           // If the type is `T | Null` or `T | Nothing`, and `T` derivesFrom the class,
           // then the OrType derivesFrom the class. Otherwise, we need to check both sides
           // derivesFrom the class.
-          if defn.isBottomType(tp.tp1) then loop(tp.tp2)
-          else loop(tp.tp1) && (defn.isBottomType(tp.tp2) || loop(tp.tp2))
+          loop(tp.tp1) && loop(tp.tp2)
+          || tp.tp1.isNullOrNothingType && loop(tp.tp2)
```

Comment:
`isBottomType` doesn't return true for `null` when `ctx.explicitNulls && !ctx.phase.erasedTypes` is true but this will, breaking null-safety.

Reply:
Right. I changed it so that `isBottomType` no longer calls `derivesFrom`.